### ISSUE TITLE
Odd behavior with Twitter/X.

### ISSUE THREAD

**Issue by hujle** [neobooru/SzuruChrome#59]
On most of the posts there that I try to upload I receive `Couldn't reverse search. Request failed with status code 504.` error. No errors neither in browser's or szurubooru instance's logs. Network configuration doesn't seem to be an issue either, because another folk of mine also facing the same issue.

**Comment by hujle** [comment]
Update: The same issue now occurs for all the posts and not only on Twitter/X, but also on Gelbooru and Deviantart. Pixiv and Danbooru, on the other hand, works perfectly fine.

**Comment by neobooru** [comment]
Can you upload those posts by manually pasting the image url on szurubooru's upload page? 
I can't directly replicate this issue, so my first guess would be that your szurubooru instance might have a timeout or something like that.

**Comment by hujle** [comment]
Nope, can't do that either. And after quite some time I am now see the error in the instance's console:
`server_1  | urllib.error.URLError: <urlopen error [Errno 110] Operation timed out>`
Now I'm starting to think the issue might have nothing to do with extension. I'll try something out later and report back if I was able to fix it or not. 👌

**Comment by hujle** [comment]
Yes, I was right. The issue had nothing to do with browser extension, but a current location of the origin server where szurubooru instance is being hosted. I was able to fix the issue by setting up a VPN tunnel between the origin server and remote server in another location, and routing all the origin server's traffic through it.

After that I was able to upload images from Twitter just fine, both through browser extension and manually pasting image's URL on upload page.
